tests/test_invenio_communities.py

```diff
@@ -274,35 +285,67 @@ def test_communities_rest_all_communities_query_and_sort(app):
-                            'id': 'comm2',
-                            'curation_policy': '',
+            assert response_data['hits'] == {
+                'hits': [
+                    {
+                        'last_record_accepted': (
+                            '2000-01-01T00:00:00+00:00'),
+                        'description': '',
+                        'title': 'A',
+                        'page': '',
+                        'id': 'comm2',
+                        'curation_policy': '',
+                        'links': {
+                            'self': ('http://inveniosoftware.org/'
+                                     'api/communities/comm2'),
+                            'html': ('http://inveniosoftware.org/'
+                                     'communities/comm2/'),
                         },
-                        {
-                            'last_record_accepted': (
-                                '2000-01-01T00:00:00+00:00'),
-                            'description': '',
-                            'title': 'B',
-                            'page': '',
-                            'id': 'comm1',
-                            'curation_policy': '',
+                    },
+                    {
+                        'last_record_accepted': (
+                            '2000-01-01T00:00:00+00:00'),
+                        'description': '',
+                        'title': 'B',
+                        'page': '',
+                        'id': 'comm1',
+                        'curation_policy': '',
+                        'links': {
+                            'self': ('http://inveniosoftware.org/'
+                                     'api/communities/comm1'),
+                            'html': ('http://inveniosoftware.org/'
+                                     'communities/comm1/'),
                         },
-                    ],
-                    'total': 2,
-                }
+                    },
+                ],
+                'total': 2,
             }
+            assert len(response_data['links']) == 1
+            assert 'sort=title' in response_data['links']['self']
+            assert 'q=comm' in response_data['links']['self']
+            assert 'size=20' in response_data['links']['self']
+            assert 'page=1' in response_data['links']['self']
 
 
 def test_communities_rest_pagination(app):
     """Test the OAI-PMH Sets creation."""
+    def check_array_in_string(items, value):
+        for item in items:
+            if item not in value:
+                return False
+        return True
+
+    def any_of(items, expression):
+        for item in items:
+            if expression(item):
+                return True
+        return False
+
+    def get_request(client, url):
+        host_size = (len(url_for('static', filename='', _external=True)) -
+                     len(url_for('static', filename='')))
+        return client.get(url[host_size:])
```

Comment:
When you make a request which also contains the host (``_external``), it was not reading the query string parameters.